machines/baseline_32_16/Makefile.machine.include

```diff
@@ -55,6 +55,9 @@ BSG_MACHINE_IO_EP_CREDITS             = 16
 BSG_MACHINE_IO_HOST_CREDITS           = $(BSG_MACHINE_IO_EP_CREDITS)
 BSG_MACHINE_IO_REMOTE_LOAD_CAP        = $(BSG_MACHINE_IO_EP_CREDITS)
 
+# (Don't) Instantiate Profilers (needed for SAIF generation, DPI Tile)
+BSG_MACHINE_NOINST_PROFILERS          = 0
```

Comment:
I don't think this should be a machine parameter, because it's not a parameter that changes hw. It should instead be a flag when building simv.

Reply:
Why not?

Reply:
I agree with @tommydcjung - where do we normally turn on profiling flags? I don't recall it being in the machine makefile.

Reply:
@mrutt92 Profiling, or trace?

Reply:
We turn on/off the tracers using a C API now, instead of using +trace.

That doesn't prevent the tracer from being instantiated. We need a solution that prevents the tracer from being instantiated.

Reply:
`ifdef ?

Reply:
Not instantiating the profilers also affects the runtime library. The library expects DPI calls to be available that return the number of FLOPS, IOPS, and cycles. If the profilers are not instantiated then the linker will fail.

Reply:
__attribute__((weak)) ?

Reply:
`ifdef would work to remove them. We could do a debug/saif solution like I mentioned above -- but we still need a solution for the C library.

Reply:
```c
__attribute__((weak))
int whatever_function_would_be_missing_if_ifdef_the_profilers_out()
{
    return NOIMPL;
}
```

Reply:
Hmmm. What if you compile both the .saif and normal version? Is there a chance the normal version can still use the weak method?

Reply:
Alternatively, we can put a macro-guarded definition in the profiler header.

Reply:
Sure. But now that I'm thinking about this, here's an argument for keeping the flag in the Makefile.machine.include instead of some random option: People know how to find/use the Makefile.machine.include.